server/src/main/java/io/crate/metadata/doc/DocIndexMetadata.java

```diff
@@ -609,6 +613,9 @@ private int columnPosition(int storedPosition) {
         columnPosition++;
 
         if (storedPosition != columnPosition) {
+            if (versionCreated.onOrAfter(Version.V_5_1_0)) {
```

Comment:
Shouldn't it be safe now to always return the stored position given that we're upgrading/ fixing the stored metadata?